File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
def solve(init: int):
    cnt = 0
    _next = init
    
    while True:
        cnt += 1
        _next = _next % 10 * 10 + (_next // 10 + _next % 10) % 10;
        
        if _next == init:
            return cnt
            
n = int(input())
print(solve(n))

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.rs
use std::io::*;

fn main() {
    let mut _in = String::new();

    stdin().read_line(&mut _in).unwrap();

    let init = _in.trim().parse::<usize>().unwrap();

    println!("{}", solve(init))
}

fn solve(init: usize) -> isize {
    let mut cnt = 0;
    let mut next = init.clone();

    loop {
        cnt += 1;
        next = next % 10 * 10 + (next / 10 + next % 10) % 10;

        if next == init {
            break;
        }
    }

    cnt
}